### PR TITLE
ENG-17024 alter persistent table

### PR DESCRIPTION
**Pull request by zeemanhe**
Export connectors are dropped while adding new column to persistent export tables because the export clause is processed after tables are parsed and hold on session.  Now the export clause is parsed while tables are created to prevent  export connector  from being dropped.

### CODE REVIEW

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 1]
```diff
@@ -1020,6 +1020,48 @@ private Statement compileAlterTableDropTTL(Table t) {
...
+        // EXPORT TO TARGET FOO ON INSERT, DELETE, UPDATE;
+        if (token.tokenType != Tokens.EXPORT) {
+            return;
+        }
+        String target = readMigrateTarget();
+        List<String> triggers = new ArrayList<>();
+        read();
+        if (token.tokenType == Tokens.ON) {
+            read();
+            while (token.tokenType != Tokens.OPENBRACKET) {
```
Comment (kaikaizi) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: The logic here seems that it also accepts syntax of `... EXPORT TO TARGET x ON INSERT DELETE (...`, where the DELETE/INSERT/UPDATE/etc. can appear multiple times.
I think the syntax is `... EXPORT TO TARGET x ON INSERT [, DELETE] [, UPDATE] (...`, and we want to guard against `... EXPORT TO TARGET x ON INSERT, INSERT ...` with same verb appearing more than once. Right?

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 2]
```diff
@@ -1020,6 +1020,48 @@ private Statement compileAlterTableDropTTL(Table t) {
...
+                } else if (token.tokenType == Tokens.UPDATEOLD) {
+                    triggers.add(Tokens.T_UPDATEOLD);
+                } else if (token.tokenType == Tokens.UPDATENEW) {
+                    triggers.add(Tokens.T_UPDATENEW);
+                } else if (token.tokenType != Tokens.COMMA){
+                    throw unexpectedToken();
+                }
+                read();
+            }
+            if (triggers.contains(Tokens.T_UPDATE) && (triggers.contains(Tokens.T_UPDATEOLD) || triggers.contains(Tokens.T_UPDATENEW))){
```
Comment (kaikaizi) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: Can you comment on an example of this?

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java [hunk 3]
```diff
@@ -1020,6 +1020,48 @@ private Statement compileAlterTableDropTTL(Table t) {
...
+                throw unexpectedToken("Cann't combine " + Tokens.T_UPDATE + " with " + Tokens.T_UPDATEOLD +
+                        " or " + Tokens.T_UPDATENEW);
+            }
+            if (triggers.contains(Tokens.T_UPDATEOLD) && triggers.contains(Tokens.T_UPDATENEW)) {
+                throw unexpectedToken("Use " + Tokens.T_UPDATE + " instead of both " + Tokens.T_UPDATEOLD +
+                        " and " + Tokens.T_UPDATENEW);
+            }
+        }
+        if (triggers.isEmpty()) {
```
Comment (kaikaizi) [inline src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: Can you comment on an example of this?

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java (hunk 2 above)
Comment (zeemanhe) [reply src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: UPDATE_OLD: export tuples before update,  UPDATE_NEW: export tuples after update

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java (hunk 3 above)
Comment (zeemanhe) [reply src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: If no triggers are found, add default trigger here

File: src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java (hunk 1 above)
Comment (zeemanhe) [reply src/hsqldb19b3/org/hsqldb_voltpatches/ParserDDL.java]: guards were added

**Pull request merged**